From c4350d53b29252d7d670d0f0e4c6bbd5ac4e0bde Mon Sep 17 00:00:00 2001
From: Cristian Prundeanu <cpru@amazon.com>
Date: Wed, 26 Mar 2025 21:52:37 +0000
Subject: [PATCH] sched/stats: Print domain name in /proc/schedstat

---
 kernel/sched/stats.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/stats.c b/kernel/sched/stats.c
index eb0cdcd4d921..ec65b7eb3ba9 100644
--- a/kernel/sched/stats.c
+++ b/kernel/sched/stats.c
@@ -138,7 +138,7 @@ static int show_schedstat(struct seq_file *seq, void *v)
 		for_each_domain(cpu, sd) {
 			enum cpu_idle_type itype;
 
-			seq_printf(seq, "domain%d %*pb", dcount++,
+			seq_printf(seq, "domain%d %s %*pb", dcount++, sd->name,
 				   cpumask_pr_args(sched_domain_span(sd)));
 			for (itype = 0; itype < CPU_MAX_IDLE_TYPES; itype++) {
 				seq_printf(seq, " %u %u %u %u %u %u %u %u",
-- 
2.34.1

